### PR TITLE
AutoClosable Stand

### PR DESCRIPTION
**Pull request by dmdashenkov**
Make Stand AutoClosable and delete `deregisterSupplierForType` method.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol PTAL
Also note, the PR build is failing. This must be an issue form `stand-in-memory`.

File: server/src/main/java/org/spine3/server/stand/Stand.java [hunk 1]
```diff
@@ -432,9 +432,12 @@ private static void feedStateRecordsToBuilder(ImmutableList.Builder<Any> resultB
...
 
-    // TODO[alex.tymchenko]: perhaps, we need to close Stand instead of doing this upon repository shutdown (see usages).
-    public void deregisterSupplierForType(TypeUrl typeUrl) {
-        typeToRepositoryMap.remove(typeUrl);
+    /**
+     * Dumps all {@link TypeUrl}-to-{@link EntityRepository} relations.
+     */
+    @Override
+    public void close() {
+        typeToRepositoryMap.clear();
```
Comment (armiol) [inline server/src/main/java/org/spine3/server/stand/Stand.java]: You need to close `StandStorage` as well.

**Comment by armiol** [comment]
@dmdashenkov the failed test is 

`org.spine3.server.stand.StandFunnelShould > deliver_updates_from_several_repositories_in_multiple_threads FAILED
    org.mockito.exceptions.verification.TooLittleActualInvocations at StandFunnelShould.java:203`

Also, please see my comment.

**Comment by dmdashenkov** [comment]
@armiol I've rebased on `stand-in-memory`, where this test is already fixed. Now the build must pass.

**Comment by dmdashenkov** [comment]
@armiol PTAL

**Comment by armiol** [comment]
@dmdashenkov LGTM.

**Pull request merged**